Fix alphanumeric_symbols regex for certain SCA remediation fields

|Wazuh version|Component|
|---|---|
| 4.4.0 | API |

## Description

During the investigation of AIT burst failures (#15344), it was found that the defined regex to validate the remediation filter in the `/sca/{agent_id}/checks/{policy_id}` API endpoint does not cover every possible case, generating failures.

Remediation example that currently fails:
<details><summary>remediation field</summary>

    "If cron will be used to schedule and run aide check: Run the following command: # crontab -u root -e Add the following line to the crontab: 0 5 * * * /usr/bin/aide.wrapper --config /etc/aide/aide.conf --check OR If aidecheck.service and aidecheck.timer will be used to schedule and run aide check: Create or edit the file /etc/systemd/system/aidecheck.service and add the following lines: [Unit] Description=Aide Check [Service] Type=simple ExecStart=/usr/bin/aide.wrapper --config /etc/aide/aide.conf --check [Install] WantedBy=multi-user.target Create or edit the file /etc/systemd/system/aidecheck.timer and add the following lines: [Unit] Description=Aide check every day at 5AM [Timer] OnCalendar=*-*-* 05:00:00 Unit=aidecheck.service [Install] WantedBy=multi-user.target Run the following commands: # chown root:root /etc/systemd/system/aidecheck.* # chmod 0644 /etc/systemd/system/aidecheck.* # systemctl daemon-reload # systemctl enable aidecheck.service # systemctl --now enable aidecheck.timer"
</details>

The regex mentioned above is assigned to the `_symbols_alphanumeric_param` variable in the `validator.py` file and later used to define the `alphanumeric_symbols` format:

https://github.com/wazuh/wazuh/blob/6ce1be2ee623eb0775355882a261d1fe495cb38b/api/api/validator.py#L17

https://github.com/wazuh/wazuh/blob/6ce1be2ee623eb0775355882a261d1fe495cb38b/api/api/validator.py#L324-L326
  
The current regex version was introduced in #10745 and the format was defined for the `remediation` filter in #5462, so we can conclude that it is not a bug introduced during the development of the `4.4` branch.

The regex has to be reviewed and modified to match every expected case and avoid errors.

## Checks
<!-- Do not modify, this will be ticked during development -->
The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

After debugging the Full test log for the failed **test_agent_PUT_endpoints.tavern.yaml** the error is in the following stage:

- Try to customly upgrade an agent selecting installer (WPK does not exist).

It was found there is an assertion error since expected response vs obtained response are opposites.
This can be seen in full_diff logged by tavern:

```
  Full diff:
    {
  -  'data': {'affected_items': [{'agent': '008',
  ?                              ^^^^^^^^^^^^^^^
  +  'data': {'affected_items': [],
  ?                              ^
  -                               'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}],
  -           'failed_items': [],
  +           'failed_items': [{'error': {'code': 1820,
  +                                       'message': 'Upgrade procedure could not '
  +                                                  'start. Agent already '
  +                                                  'upgrading'},
  +                             'id': ['008']}],
  -           'total_affected_items': 1,
  ?                                   ^
  +           'total_affected_items': 0,
  ?                                   ^
  -           'total_failed_items': 0},
  ?                                 ^
  +           'total_failed_items': 1},
  ?                                 ^
  -  'message': <tavern.util.loader.StrSentinel object at 0x7f55e4a49df0>,
  +  'error': 1,
  +  'message': 'No upgrade task was created',
    }
```

But after testing locally, the error seems to be only when executed by tavern.

Local test info:

```
Local endpoint **/agents/upgrade_custom?agents_list=008&installer=upgrade.sh&file_path=/var/ossec/no_existant.wpk**  test response:

{
    "data": {
        "affected_items": [
            {
                "agent": "008",
                "task_id": 9
            }
        ],
        "total_affected_items": 1,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "All upgrade tasks were created",
    "error": 0
}
```

Tavern Full test log: 

```
=================================== FAILURES ===================================
_ /home/eduardoleon/git/wazuh/api/test/integration/test_agent_PUT_endpoints.tavern.yaml::PUT /agents/{agent_id}/upgrade_custom _
Format variables:
  protocol:s = 'https'
  host:s = 'localhost'
  port:d = '55000'
  test_login_token = 'eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjc4MjAwMzExLCJleHAiOjE2NzgyMDEyMTEsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.ACA5cbDwOgqYZrQfEZiwlXrft_Dt0HQAJwkM_zuE0XGpVb8yZ8QFzs6MAkm8VYRlkfnxTbTOji1tpsvw1Hb8oeg2AdlCQTE5P5w6ZKLzLled_1iT4uXV47uy2D8VRHw8G99VH9upVrVTxVpJOwqAOM_Sh2q9HKCQb3EP5qJWOrZNGiCO'
  custom_wpk_path:s = '/var/ossec/test.wpk'

Source test stage (line 1237):
  - name: Try to customly upgrade an agent selecting installer (WPK does not exist)
    request:
      verify: False
      url: "{protocol:s}://{host:s}:{port:d}/agents/upgrade_custom"
      method: PUT
      headers:
        Authorization: "Bearer {test_login_token}"
      params:
        agents_list: '008'
        installer: upgrade.sh
        file_path: "{custom_wpk_path:s}"
    response:
      status_code: 200
      json:
        data:
          affected_items:
            - agent: '008'
              task_id: !anyint
          total_affected_items: 1
          total_failed_items: 0
          failed_items: [ ]
        message: !anystr


Formatted stage:
  name: Try to customly upgrade an agent selecting installer (WPK does not exist)
  request:
    headers:
      Authorization: Bearer eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjc4MjAwMzExLCJleHAiOjE2NzgyMDEyMTEsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.ACA5cbDwOgqYZrQfEZiwlXrft_Dt0HQAJwkM_zuE0XGpVb8yZ8QFzs6MAkm8VYRlkfnxTbTOji1tpsvw1Hb8oeg2AdlCQTE5P5w6ZKLzLled_1iT4uXV47uy2D8VRHw8G99VH9upVrVTxVpJOwqAOM_Sh2q9HKCQb3EP5qJWOrZNGiCO
    method: PUT
    params:
      agents_list: 008
      file_path: '/var/ossec/test.wpk'
      installer: upgrade.sh
    url: 'https://localhost:55000/agents/upgrade_custom'
    verify: false
  response:
    json:
      data:
        affected_items:
        - agent: 008
          task_id: !anyint ''
        failed_items: []
        total_affected_items: 1
        total_failed_items: 0
      message: !anystr ''
    status_code: 200

Errors:
E   tavern.util.exceptions.TestFailError: Test 'Try to customly upgrade an agent selecting installer (WPK does not exist)' failed:
    - Key mismatch: (expected["data"]["total_failed_items"] = '0' (type = <class 'int'>), actual["data"]["total_failed_items"] = '1' (type = <class 'int'>))
------------------------------ Captured log call -------------------------------
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '400'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
ERROR    tavern.response.base:base.py:41 Key mismatch: (expected["data"]["total_failed_items"] = '0' (type = <class 'int'>), actual["data"]["total_failed_items"] = '1' (type = <class 'int'>))
Traceback (most recent call last):
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert {'data': {'affected_items': [],\n          'failed_items': [{'error': {'code': 1820,\n                                      'message': 'Upgrade procedure could not '\n                                                 'start. Agent already '\n                                                 'upgrading'},\n                            'id': ['008']}],\n          'total_affected_items': 0,\n          'total_failed_items': 1},\n 'error': 1,\n 'message': 'No upgrade task was created'} == {'data': {'affected_items': [{'agent': '008',\n                              'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}],\n          'failed_items': [],\n          'total_affected_items': 1,\n          'total_failed_items': 0},\n 'message': <tavern.util.loader.StrSentinel object at 0x7f55e4a49df0>}
  Differing items:
  {'data': {'affected_items': [], 'failed_items': [{'error': {'code': 1820, 'message': 'Upgrade procedure could not start. Agent already upgrading'}, 'id': ['008']}], 'total_affected_items': 0, 'total_failed_items': 1}} != {'data': {'affected_items': [{'agent': '008', 'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}], 'failed_items': [], 'total_affected_items': 1, 'total_failed_items': 0}}
  {'message': 'No upgrade task was created'} != {'message': <tavern.util.loader.StrSentinel object at 0x7f55e4a49df0>}
  Left contains 1 more item:
  {'error': 1}
  Full diff:
    {
  -  'data': {'affected_items': [{'agent': '008',
  ?                              ^^^^^^^^^^^^^^^
  +  'data': {'affected_items': [],
  ?                              ^
  -                               'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}],
  -           'failed_items': [],
  +           'failed_items': [{'error': {'code': 1820,
  +                                       'message': 'Upgrade procedure could not '
  +                                                  'start. Agent already '
  +                                                  'upgrading'},
  +                             'id': ['008']}],
  -           'total_affected_items': 1,
  ?                                   ^
  +           'total_affected_items': 0,
  ?                                   ^
  -           'total_failed_items': 0},
  ?                                 ^
  +           'total_failed_items': 1},
  ?                                 ^
  -  'message': <tavern.util.loader.StrSentinel object at 0x7f55e4a49df0>,
  +  'error': 1,
  +  'message': 'No upgrade task was created',
    }

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert {'affected_items': [],\n 'failed_items': [{'error': {'code': 1820,\n                             'message': 'Upgrade procedure could not start. '\n                                        'Agent already upgrading'},\n                   'id': ['008']}],\n 'total_affected_items': 0,\n 'total_failed_items': 1} == {'affected_items': [{'agent': '008',\n                     'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}],\n 'failed_items': [],\n 'total_affected_items': 1,\n 'total_failed_items': 0}
  Differing items:
  {'total_failed_items': 1} != {'total_failed_items': 0}
  {'total_affected_items': 0} != {'total_affected_items': 1}
  {'affected_items': []} != {'affected_items': [{'agent': '008', 'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}]}
  {'failed_items': [{'error': {'code': 1820, 'message': 'Upgrade procedure could not start. Agent already upgrading'}, 'id': ['008']}]} != {'failed_items': []}
  Full diff:
    {
  -  'affected_items': [{'agent': '008',
  -                      'task_id': <tavern.util.loader.IntSentinel object at 0x7f55e4a49eb0>}],
  -  'failed_items': [],
  ?    ^^^
  +  'affected_items': [],
  ?   + ^^^^
  +  'failed_items': [{'error': {'code': 1820,
  +                              'message': 'Upgrade procedure could not start. '
  +                                         'Agent already upgrading'},
  +                    'id': ['008']}],
  -  'total_affected_items': 1,
  ?                          ^
  +  'total_affected_items': 0,
  ?                          ^
  -  'total_failed_items': 0,
  ?                        ^
  +  'total_failed_items': 1,
  ?                        ^
    }

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert 1 == 0
  +1
  -0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/response/base.py", line 88, in recurse_check_key_match
    check_keys_match_recursive(expected_block, block, [], strict)
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 462, in check_keys_match_recursive
    check_keys_match_recursive(
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 462, in check_keys_match_recursive
    check_keys_match_recursive(
  File "/home/eduardoleon/.pyenv/versions/3.9.9/envs/inttegration-3.9-venv/lib/python3.9/site-packages/tavern/util/dict_util.py", line 546, in check_keys_match_recursive
    raise exceptions.KeyMismatchError(
tavern.util.exceptions.KeyMismatchError: Key mismatch: (expected["data"]["total_failed_items"] = '0' (type = <class 'int'>), actual["data"]["total_failed_items"] = '1' (type = <class 'int'>))
=========================== short test summary info ============================
FAILED test_agent_PUT_endpoints.tavern.yaml::PUT /agents/{agent_id}/upgrade_custom

```


## Next steps:
- [ ] Discuss with the team the on how to address the tavern assertion error.

--- a/api/api/validator.py
+++ b/api/api/validator.py
@@ -14,7 +14,7 @@ from wazuh.core import common
 from wazuh.core.exception import WazuhError
 
 _alphanumeric_param = re.compile(r'^[\w,\-.+\s:]+$')
-_symbols_alphanumeric_param = re.compile(r'^[\w,<>!\-.+\s:/()\[\]\'\"|=~#]+$')
+_symbols_alphanumeric_param = re.compile(r'^[\w,*<>!\-.+\s:/()\[\]\'\"|=~#]+$')
 _array_numbers = re.compile(r'^\d+(,\d+)*$')
 _array_names = re.compile(r'^[\w\-.%]+(,[\w\-.%]+)*$')
 _base64 = re.compile(r'^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$')

